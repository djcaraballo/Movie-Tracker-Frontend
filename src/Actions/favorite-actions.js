export const addFavorite = (movie, user) => ({
  type: 'ADD_FAVORITE',
  movie_id: movie.id,
  user_id: user.id,
  title: movie.title,
  poster_path: movie.poster_path,
  release_date: movie.release_date,
  vote_average: movie.vote_average,
  overview: movie.overview,
});