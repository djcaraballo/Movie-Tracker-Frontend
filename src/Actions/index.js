export const addMovies = (movieInfo) => ({
  type: 'ADD_MOVIES',
  movies: movieInfo
})