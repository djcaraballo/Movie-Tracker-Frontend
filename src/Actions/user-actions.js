export const newUser = (name, email, password) => ({
  type: 'NEW_USER',
  name,
  email,
  password,
});

export const fetchUser = (status) => ({
  type: 'FETCH_USER',
  status,
});

export const toggleUserStatus = (status) => ({
  type: 'TOGGLE_STATUS',
  status: !status,
});
