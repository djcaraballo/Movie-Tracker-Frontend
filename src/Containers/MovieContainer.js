import React from 'react';
import PropTypes from 'prop-types';

const MovieContainer = () => {
  
  return (
    <div className='movie-container'>
      
    </div>
  )
}

MovieContainer.propTypes = {

}

export default MovieContainer