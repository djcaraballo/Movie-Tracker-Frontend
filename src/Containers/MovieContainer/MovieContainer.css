.movie-container {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr)); 
  grid-row-gap: 3rem;
  grid-column-gap: 1rem;
  margin: 1rem 2rem;
}

.movie-container > div {
  border-bottom: double #01838c;
}
