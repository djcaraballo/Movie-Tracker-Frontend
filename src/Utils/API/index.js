import { key } from './API_KEY';

export const fetchMovies = async() => {
  const url = `https://api.themoviedb.org/3/movie/now_playing?api_key=${key}&language=en-US&page=1`;
  const response = await fetch(url);
  const result = await response.json();
  return result;
}

export const postNewUser = async (newUserInfo) => {
  fetch('http://localhost:3000/api/users/new',{ 
    method: "POST",
    body: JSON.stringify({...newUserInfo, favorites: []}),
    headers:{'Content-Type': 'application/json'},
  })
  .then(res => res.json())
  .then(response => console.log('Success:', JSON.stringify(response)))
  .catch(error => console.error('Error:', error));
}

export const signInUser= async (userInfo) => {
  fetch('http://localhost:3000/api/users',{
	method: "POST",
	body: JSON.stringify(userInfo),
	headers:{'Content-Type': 'application/json'},
  })
  .then(res => res.json())
  .then(response => console.log('Success:', JSON.stringify(response)))
  .catch(error => console.error('Error:', error));
}
