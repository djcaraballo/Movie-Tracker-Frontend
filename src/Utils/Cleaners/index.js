import * as API from '../API/';

export const cleanMovieData = async () => {
  const rawMovieData = await API.fetchMovies()
  const posterURL = 'https://image.tmdb.org/t/p/w300_and_h450_bestv2';
  const backdropURL = 'http://image.tmdb.org/t/p/w1280';
  return rawMovieData.results.map(movie => {
    const {title, id, poster_path, backdrop_path, release_date, vote_average, overview} = movie;
    
    return {
      title, 
      id, 
      poster_path: `${posterURL}${poster_path}`, 
      backdrop_path: `${backdropURL}${backdrop_path}`,
      release_date, 
      vote_average, 
      overview
    }
  });
}


