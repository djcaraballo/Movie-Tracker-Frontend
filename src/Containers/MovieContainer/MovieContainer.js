import React, { Component } from "react";
import { connect } from "react-redux";
import { NavLink, Route } from "react-router-dom";
import { filterMovies, addMovies, clearMovies } from "../../Actions/";
import * as DataCleaner from "../../Utils/Cleaners/";
import "./MovieContainer.css";
import MovieCard from "../../Components/MovieCard/MovieCard";

export class MovieContainer extends Component {
  resetMovies = async () => {
    const movieInfo = await DataCleaner.cleanMovieData();
    this.props.clearMovies();
    this.props.addMovies(movieInfo);
  };

  render() {
    const cards = this.props.movies.map(movie => {
      return <MovieCard key={movie.id} {...movie} />;
    });

    return (
      <div>
        <Route
          exact
          path="/favorites"
          render={() => (
            <div>
              <div className="nav-btn-container">
                <NavLink to="/" className="home-btn" onClick={this.resetMovies}>
                  HOME
                </NavLink>
              </div>
              <div className="movie-container">{cards}</div>
            </div>
          )}
        />
        <Route
          exact
          path="/"
          render={() => (
            <div>
              <div className="nav-btn-container">
                <NavLink
                  to="/favorites"
                  className="favorites-btn"
                  onClick={() => this.props.updateMovies(this.props.movies)}
                >
                  FAVORITES
                </NavLink>
              </div>
              <div className="movie-container">{cards}</div>
            </div>
          )}
        />
      </div>
    );
  }
}

export const mapStateToProps = state => {
  return {
    movies: state.movies,
    user: state.user
  };
};

export const mapDispatchToProps = dispatch => ({
  updateMovies: movies => dispatch(filterMovies(movies)),
  addMovies: movieInfo => dispatch(addMovies(movieInfo)),
  clearMovies: () => dispatch(clearMovies())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MovieContainer);
