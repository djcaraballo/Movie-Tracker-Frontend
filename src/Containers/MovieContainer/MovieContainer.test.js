import React from 'react';
import ReactDOM from 'react-dom';
import { MovieContainer, mapStateToProps, mapDispatchToProps} from './MovieContainer';
import { shallow } from 'enzyme';
import * as mockDataCleaner from '../../Utils/Cleaners/';
import { addMovies, filterMovies, clearMovies } from '../../Actions/';

describe('MovieContainer', () => {
  let wrapper;
  const mockUser = {
    name: 'Louisa',
    id: 1806,
    email: 'Louisa@turing.io',
    password: 'oooooooooooookay'
  };
  const mockMovies = [
    {
      backdrop_path:
        'http://image.tmdb.org/t/p/w1280/VuukZLgaCrho2Ar8Scl9HtV3yD.jpg',
      id: 335983,
      overview:
        'When Eddie Brock acquires the powers of a symbiote, he will have to release his alter-ego "Venom" to save his life.',
      poster_path:
        'https://image.tmdb.org/t/p/w300_and_h450_bestv2/2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg',
      release_date: '2018-10-03',
      title: 'Venom',
      vote_average: 6.6,
      favorite: false,
    },
    {
      backdrop_path:
        'http://image.tmdb.org/t/p/w1280/840rbblaLc4SVxm8gF3DNdJ0YAE.jpg',
      id: 332562,
      overview:
        "Seasoned musician Jackson Maine discovers—and falls in love with—struggling artist Ally. She has just about given up on her dream to make it big as a singer—until Jack coaxes her into the spotlight. But even as Ally's career takes off, the personal side of their relationship is breaking down, as Jack fights an ongoing battle with his own internal demons.",
      poster_path:
        'https://image.tmdb.org/t/p/w300_and_h450_bestv2/wrFpXMNBRj2PBiN4Z5kix51XaIZ.jpg',
      release_date: '2018-10-03',
      title: 'A Star Is Born',
      vote_average: 7.5,
      favorite: false,
    },
    {
      backdrop_path:
        'http://image.tmdb.org/t/p/w1280/f4E0ocYeToEuXvczZv6QArrMDJ.jpg',
      id: 346910,
      overview:
        'From the outer reaches of space to the small-town streets of suburbia, the hunt comes home. Now, the universe’s most lethal hunters are stronger, smarter and deadlier than ever before, having genetically upgraded themselves with DNA from other species. When a young boy accidentally triggers their return to Earth, only a ragtag crew of ex-soldiers and a disgruntled science teacher can prevent the end of the human race.',
      poster_path:
        'https://image.tmdb.org/t/p/w300_and_h450_bestv2/wMq9kQXTeQCHUZOG4fAe5cAxyUA.jpg',
      release_date: '2018-09-13',
      title: 'The Predator',
      vote_average: 5.3,
      favorite: false,
    },
  ];
  const mockMovieInfo = {
        backdrop_path:
          'http://image.tmdb.org/t/p/w1280/f4E0ocYeToEuXvczZv6QArrMDJ.jpg',
        id: 346910,
        overview:
          'From the outer reaches of space to the small-town streets of suburbia, the hunt comes home. Now, the universe’s most lethal hunters are stronger, smarter and deadlier than ever before, having genetically upgraded themselves with DNA from other species. When a young boy accidentally triggers their return to Earth, only a ragtag crew of ex-soldiers and a disgruntled science teacher can prevent the end of the human race.',
        poster_path:
          'https://image.tmdb.org/t/p/w300_and_h450_bestv2/wMq9kQXTeQCHUZOG4fAe5cAxyUA.jpg',
        release_date: '2018-09-13',
        title: 'The Predator',
        vote_average: 5.3,
        favorite: false,
      }
  const mockUpdateMovies = jest.fn();
  const mockAddMovies = jest.fn();
  const mockClearMovies = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <MovieContainer
        movies={mockMovies}
        user={mockUser}
        updateMovies={mockUpdateMovies}
        addMovies={mockAddMovies}
        clearMovies={mockClearMovies} />)
  })
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('resetMovies function', () => {
    it('should call cleanMovieData', async () => {
      mockDataCleaner.cleanMovieData = jest.fn()
      wrapper.instance().resetMovies()
      await expect(mockDataCleaner.cleanMovieData).toHaveBeenCalled()
    })

    it('should call clearMovies', async () => {
      wrapper.instance().resetMovies()
      await expect(mockClearMovies).toHaveBeenCalled()
    })

    it('should call addMovies with the correct params', async () => {
      mockDataCleaner.cleanMovieData = jest.fn().mockImplementation(() => {
        Promise.resolve(mockMovieInfo)
      })
      wrapper.instance().resetMovies()
      await expect(mockAddMovies).toHaveBeenCalled()
    })
  })

  describe('mapStateToProps function', () => {
    it('should return an object with a user and a movies array', () => {
      const mockState = {
        movies: [
          {title: 'Coco', vote_average: 7.6, favorite: true},
          {title: 'Inception', vote_average: 8.6, favorite: true},
          {title: 'Hunt for Red October', vote_average: 4.6, favorite: true}
        ],
        user: {name: 'Louisa', email: 'blah@gmail.com'},
        favorites: []
      }
      const expected = {
        user: {name: 'Louisa', email: 'blah@gmail.com'},
        movies: [
          {title: 'Coco', vote_average: 7.6, favorite: true},
          {title: 'Inception', vote_average: 8.6, favorite: true},
          {title: 'Hunt for Red October', vote_average: 4.6, favorite: true}
        ],
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps function', () => {
    const mockDispatch = jest.fn()
    const mockFilterAction = filterMovies(mockMovies)
    const mockAddAction = addMovies(mockMovieInfo)
    const mockClearAction = clearMovies()
    let mappedProps;

    beforeEach(() => {
      mappedProps = mapDispatchToProps(mockDispatch)
    })

    it('should call dispatch with a filterMovies action when updateMovies is called', () => {
      mappedProps.updateMovies(mockMovies)
      expect(mockDispatch).toHaveBeenCalledWith(mockFilterAction)
    })

    it('should call dispatch with an addMovies action when addMovies is called', () => {
      mappedProps.addMovies(mockMovieInfo)
      expect(mockDispatch).toHaveBeenCalledWith(mockAddAction)
    })

    it('should call dispatch with a clearMovies action when clearMovies is called', () => {
      mappedProps.clearMovies()
      expect(mockDispatch).toHaveBeenCalledWith(mockClearAction)
    })
  })
});
