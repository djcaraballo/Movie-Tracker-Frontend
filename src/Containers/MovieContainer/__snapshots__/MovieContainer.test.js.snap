// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(MovieContainer)
    store={
      Object {
        "dispatch": [MockFunction],
        "getState": [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": Object {
                "movies": Array [
                  Object {
                    "backdrop_path": "/somethingelse.jpg",
                    "id": 9,
                    "overview": "This is a movie.",
                    "poster_path": "/something.jpg",
                    "release_date": "2018",
                    "title": "Venom",
                    "vote_average": 6.6,
                  },
                ],
              },
            },
            Object {
              "isThrow": false,
              "value": Object {
                "movies": Array [
                  Object {
                    "backdrop_path": "/somethingelse.jpg",
                    "id": 9,
                    "overview": "This is a movie.",
                    "poster_path": "/something.jpg",
                    "release_date": "2018",
                    "title": "Venom",
                    "vote_average": 6.6,
                  },
                ],
              },
            },
          ],
        },
        "subscribe": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "dispatch": [MockFunction],
      "movies": Array [
        Object {
          "backdrop_path": "/somethingelse.jpg",
          "id": 9,
          "overview": "This is a movie.",
          "poster_path": "/something.jpg",
          "release_date": "2018",
          "title": "Venom",
          "vote_average": 6.6,
        },
      ],
      "store": Object {
        "dispatch": [MockFunction],
        "getState": [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": Object {
                "movies": Array [
                  Object {
                    "backdrop_path": "/somethingelse.jpg",
                    "id": 9,
                    "overview": "This is a movie.",
                    "poster_path": "/something.jpg",
                    "release_date": "2018",
                    "title": "Venom",
                    "vote_average": 6.6,
                  },
                ],
              },
            },
            Object {
              "isThrow": false,
              "value": Object {
                "movies": Array [
                  Object {
                    "backdrop_path": "/somethingelse.jpg",
                    "id": 9,
                    "overview": "This is a movie.",
                    "poster_path": "/something.jpg",
                    "release_date": "2018",
                    "title": "Venom",
                    "vote_average": 6.6,
                  },
                ],
              },
            },
          ],
        },
        "subscribe": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [MockFunction],
          "getState": [MockFunction] {
            "calls": Array [
              Array [],
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": Object {
                  "movies": Array [
                    Object {
                      "backdrop_path": "/somethingelse.jpg",
                      "id": 9,
                      "overview": "This is a movie.",
                      "poster_path": "/something.jpg",
                      "release_date": "2018",
                      "title": "Venom",
                      "vote_average": 6.6,
                    },
                  ],
                },
              },
              Object {
                "isThrow": false,
                "value": Object {
                  "movies": Array [
                    Object {
                      "backdrop_path": "/somethingelse.jpg",
                      "id": 9,
                      "overview": "This is a movie.",
                      "poster_path": "/something.jpg",
                      "release_date": "2018",
                      "title": "Venom",
                      "vote_average": 6.6,
                    },
                  ],
                },
              },
            ],
          },
          "subscribe": [MockFunction] {
            "calls": Array [
              Array [
                [Function],
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
        },
        "unsubscribe": undefined,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "dispatch": [MockFunction],
        "movies": Array [
          Object {
            "backdrop_path": "/somethingelse.jpg",
            "id": 9,
            "overview": "This is a movie.",
            "poster_path": "/something.jpg",
            "release_date": "2018",
            "title": "Venom",
            "vote_average": 6.6,
          },
        ],
        "store": Object {
          "dispatch": [MockFunction],
          "getState": [MockFunction] {
            "calls": Array [
              Array [],
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": Object {
                  "movies": Array [
                    Object {
                      "backdrop_path": "/somethingelse.jpg",
                      "id": 9,
                      "overview": "This is a movie.",
                      "poster_path": "/something.jpg",
                      "release_date": "2018",
                      "title": "Venom",
                      "vote_average": 6.6,
                    },
                  ],
                },
              },
              Object {
                "isThrow": false,
                "value": Object {
                  "movies": Array [
                    Object {
                      "backdrop_path": "/somethingelse.jpg",
                      "id": 9,
                      "overview": "This is a movie.",
                      "poster_path": "/something.jpg",
                      "release_date": "2018",
                      "title": "Venom",
                      "vote_average": 6.6,
                    },
                  ],
                },
              },
            ],
          },
          "subscribe": [MockFunction] {
            "calls": Array [
              Array [
                [Function],
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [MockFunction],
            "getState": [MockFunction] {
              "calls": Array [
                Array [],
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": Object {
                    "movies": Array [
                      Object {
                        "backdrop_path": "/somethingelse.jpg",
                        "id": 9,
                        "overview": "This is a movie.",
                        "poster_path": "/something.jpg",
                        "release_date": "2018",
                        "title": "Venom",
                        "vote_average": 6.6,
                      },
                    ],
                  },
                },
                Object {
                  "isThrow": false,
                  "value": Object {
                    "movies": Array [
                      Object {
                        "backdrop_path": "/somethingelse.jpg",
                        "id": 9,
                        "overview": "This is a movie.",
                        "poster_path": "/something.jpg",
                        "release_date": "2018",
                        "title": "Venom",
                        "vote_average": 6.6,
                      },
                    ],
                  },
                },
              ],
            },
            "subscribe": [MockFunction] {
              "calls": Array [
                Array [
                  [Function],
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            },
          },
          "unsubscribe": undefined,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
